transport/src/main/java/io/netty/channel/FileDescriptor.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright 2015 The Netty Project
+ *
+ * The Netty Project licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package io.netty.channel;
+
+import java.io.IOException;
+
+public interface FileDescriptor {
+
+    /**
+     * An invalid file descriptor which was closed before.
+     */
+    FileDescriptor INVALID = new FileDescriptor() {
+        @Override
+        public int intValue() {
+            return -1;
+        }
+
+        @Override
+        public void close() {
+            // MOOP
+        }
+    };
+
+    /**
+     * Return the int value of the filedescriptor.
+     */
+    int intValue();
+
+    /**
+     * Close the file descriptor.
+     */
+    void close() throws IOException;
```

Comment:
I guess then it is easy to be overlooked. Well, let's just leave it as it is for now.